runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -754,6 +754,7 @@ handleServerMessage(JITServer::ClientStream *client, TR_J9VM *fe, JITServer::Mes
 
          void *thunkAddress = reinterpret_cast<void *>(thunkStart);
          fe->setInvokeExactJ2IThunk(thunkAddress, comp);
+         compInfoPT->reloRuntime()->reloTarget()->flushCache(reinterpret_cast<uint8_t *>(thunkAddress), serializedThunk.size());
```

Comment:
I am not sure if it matters, but should the `flushCache` be done immediately after copying the code to the code cache and before creating the mapping signature-->thunkCode in `fe->setInvokeExactJ2IThunk(thunkAddress, comp);`

Reply:
I moved `flushCache` call before `fe->setInvokeExactJ2IThunk(thunkAddress, comp);` .

Reply:
I'm curious why this wasn't written as
```c
         compInfoPT->reloRuntime()->reloTarget()->flushCache(thunkStart, serializedThunk.size());
```
which doesn't need the `reinterpret_cast`?

Reply:
Ah, I did not notice that.
I copied `compInfoPT->reloRuntime()->reloTarget()->performThunkRelocation(reinterpret_cast<uint8_t *>(thunkAddress), (uintptr_t)vmHelper);` from `VM_setJ2IThunk` case and modified it. I did not notice that I could use `thunkStart` as is for `VM_setInvokeExactJ2IThunk`.

Reply:
I opened https://github.com/eclipse-openj9/openj9/pull/17914 to address it.